mk/spksrc.service.installer.dsm7

```diff
@@ -215,10 +232,12 @@ preupgrade ()
     call_func "validate_preupgrade"
 
     # dsm6 -> dsm7
-    # Migrate data to permanent storage
+    # Backup & Migrate data once
+    # to permanent storage
     if [ -d ${SYNOPKG_PKGDEST}/var -a ! "$(ls -A ${SYNOPKG_PKGVAR})" ]; then
-        # only migrate when destination is empty
-        $CP ${SYNOPKG_PKGDEST}/var/. ${SYNOPKG_PKGVAR} 2>&1 | install_log
+        $RSYNC --ignore-existing ${SYNOPKG_PKGDEST}/var/ ${SYNOPKG_PKGVAR} 2>&1 | install_log && chk=$?
```

Comment:
@th0ma7 Sorry for the long delay to my response. Reading back some conversation I misunderstood the usage of the `@ `. Ups (Using the same folder names as Synology would course confusion later on anyway, so I'm happy you renamed them.)

Regarding rsync would you be able to copy the rsync command to a new PR? I think this can be merged fairly quickly.

Overall feel that maybe that there could be a better implementation, but I'm not sure. I think If we want this we could move slowly to where we can implement some of this one folder at a time (every folder here is special in their own way and needs different handling). 

IMHO The home folder is of interest because of the restrictive permissions, and it's actually defined as an environment variable that some packages can make use of (`$HOME`) and/or is the default working directory (I can't remember exactly). Packages just need to update their scrips, so I don't think the framework needs any changes here. Supporting `etc/` would be nice, but I don't think there is any rush to do so. 

Let me know what you think.

Reply:
> I misunderstood the usage of the `@ `. Ups (Using the same folder syntax as Synology would course confusion later on anyway, so I'm happy you renamed them.)

Currently using `_` instead.  @hgy59 also had issues with it.  As long as we are ok with one delimiter.  In the meantime I'll stick with `_`.

> Regarding rsync would you be able to copy the rsync command to a new PR? I think this can be merged fairly quickly.

Certainly, will do.

> Overall feel that maybe that there could be a better implementation, but I'm not sure. I think If we want this we could move slowly to where we can implement some of this one folder at a time (every folder here is special in their own way and needs different handling).

I could start off with proposed adjustments to `var` leaving room for others after.  To fully work it must include changes to the default GNU configuration options  such as `--localstatedir=$(INSTALL_PREFIX_VAR)` (and also CMAKE currently missing from PR).  I'll create a second PR for added adjustments specific to `var`.

> IMHO The home folder is of interest because of the restrictive permissions, and it's actually defined as an environment variable that some packages can make use of (`$HOME`) and/or is the default working directory (I can't remember exactly). Packages just need to update their scrips, so I don't think the framework needs any changes here. Supporting `etc/` would be nice, but I don't think there is any rush to do so.

Approach looks good to me.  Lets hit `rsync` + adjustments to `var` first.  See how things goes then perhaps add `etc` and/or `home` as we see fit (and perhaps `tmp` but probably less needed).  In the meantime I'll keep this PR in sync with master as things moves forward.
 
> Let me know what you think.

Thnx.  And btw, will be mostly offline for next weeks.